app/models/devise_token_auth/concerns/user.rb

```diff
@@ -218,13 +218,8 @@ def destroy_expired_tokens
   end
 
   def should_remove_tokens_after_password_reset?
-    if Rails::VERSION::MAJOR <= 5 ||defined?('Mongoid')
-      encrypted_password_changed? &&
+    encrypted_password_changed? &&
         DeviseTokenAuth.remove_tokens_after_password_reset
-    else
-      saved_change_to_attribute?(:encrypted_password) &&
```

Comment:
`saved_change_to_attribute?`
- will return false in `before_save` callback
- will return true in `after_save` callback

But `encrypted_password_changed?` is opposite to `saved_change_to_attribute?`
So we must use `encrypted_password_changed?`